controllers/control/cassandratask_controller.go

```diff
@@ -273,7 +275,17 @@ func (r *CassandraTaskReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		case api.CommandCleanup:
 			cleanup(taskConfig)
 		case api.CommandRestart:
-			r.restart(taskConfig)
+			// This job is targeting StatefulSets and not Pods
+			sts, err := r.getDatacenterStatefulSets(ctx, dc)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+
+			res, err = r.restartSts(ctx, sts, taskConfig)
```

Comment:
The `res` value doesn't seem to be used at all. Maybe that's what could be causing prematurely setting the completion time?

Reply:
Isn't that res used on the line 333? I don't see it being overwritten anywhere.

Reply:
Isn't it overwritten on line 320?

```
res, failed, completed, err = r.reconcileEveryPodTask(ctx, dc, taskConfig)
```

Reply:
Or is `break JobDefinition` sending us straight to line 333?

Reply:
break JobDefinition should break us from the for() clause and thus line 320 should never be run.